tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedTargetedFailPointOperation.cs

```diff
@@ -0,0 +1,82 @@
+﻿/* Copyright 2020-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using MongoDB.Bson;
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.TestHelpers;
+
+namespace MongoDB.Driver.Tests.UnifiedTestOperations
+{
+    public class UnifiedTargetedFailPointOperation : IUnifiedFailPointOperation
+    {
+        private readonly BsonDocument _failPointCommand;
+        private readonly IClientSessionHandle _session;
+
+        public UnifiedTargetedFailPointOperation(
+            IClientSessionHandle session,
+            BsonDocument failPointCommand)
+        {
+            _session = session;
+            _failPointCommand = failPointCommand;
+        }
+
+        public void Execute(out FailPoint failPoint)
+        {
+            var pinnedServer = _session?.WrappedCoreSession?.CurrentTransaction?.PinnedServer;
+            if (pinnedServer == null)
+            {
+                throw new InvalidOperationException("UnifiedTargetedFailPointOperation requires a pinned server");
```

Comment:
This exception was missed when adding periods to the end of exception messages:

    throw new InvalidOperationException("UnifiedTargetedFailPointOperation requires a pinned server.");

Reply:
Done